trillian/ctfe/ct_server/main.go

```diff
@@ -185,3 +206,37 @@ func awaitSignal(doneFn func()) {
 
 	doneFn()
 }
+
+func setupAndRegister(ctx context.Context, client trillian.TrillianLogClient, deadline time.Duration, cfg *configpb.LogConfig) error {
+	handlers, err := ctfe.SetUpInstance(ctx, client, cfg, deadline, prometheus.MetricFactory{})
+	if err != nil {
+		return err
+	}
+	for path, handler := range *handlers {
+		http.Handle(path, handler)
+	}
+	return nil
+}
+
+func readMultiCfg(filename string) (*configpb.LogMultiConfig, map[string]*configpb.LogBackend, error) {
+	cfg, err := ctfe.MultiLogConfigFromFile(filename)
+	if err != nil {
+		return nil, nil, err
+	}
+	beMap, err := ctfe.ValidateLogMultiConfig(cfg)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return cfg, beMap, nil
+}
+
+func readCfg(filename string, backendSpec string) (*configpb.LogMultiConfig, map[string]*configpb.LogBackend, error) {
+	cfg, err := ctfe.LogConfigFromFile(filename)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	mCfg, beMap := ctfe.ToMultiLogConfig(cfg, backendSpec)
```

Comment:
Idle thought: might be worth having `ToMultiLogConfig` only build the `LogMultiConfig`, not the map, and then have the `ValidateLogMultiConfig` & map-building pulled up into the mainline and run for either case.  (Or not, up to you.)

Reply:
Done. I think it's better to have ToMultiLogConfig do a single job.